src/OrderByAggregatesPlugin.ts

```diff
@@ -158,6 +159,70 @@ const OrderByAggregatesPlugin: Plugin = (builder) => {
         });
       });
 
+      // proc aggregates
+      pgAggregateSpecs.forEach((spec) => {
+        introspectionResultsByKind.procedure.forEach((proc: PgProc) => {
+          if (proc.returnsSet) return
+
+          const type = introspectionResultsByKind.typeById[proc.returnTypeId];
+          if (
+            (spec.shouldApplyToEntity && !spec.shouldApplyToEntity(proc)) ||
+            !spec.isSuitableType(type)
+          ) return;
+          const computedColumnDetails = getComputedColumnDetails(
+            build,
+            table,
+            proc
+          );
+          if (!computedColumnDetails) return;
+          const { pseudoColumnName } = computedColumnDetails;
+          const fieldName = inflection.computedColumn(
+            pseudoColumnName,
+            proc,
+            table
+          );
+          
+          memo = build.extend(
+            memo,
+            orderByAscDesc(
+              inflection.orderByProcAggregateOfManyRelationByKeys(
+                keys,
+                table,
+                foreignTable,
+                constraint,
+                spec,
+                fieldName
+              ),
+              ({ queryBuilder }) => {
+                const foreignTableAlias = queryBuilder.getTableAlias();
+                const conditions: SQL[] = [];
+                keys.forEach((key, i) => {
+                  conditions.push(
+                    sql.fragment`${tableAlias}.${sql.identifier(
+                      key.name
+                    )} = ${foreignTableAlias}.${sql.identifier(
+                      foreignKeys[i].name
+                    )}`
+                  );
+                });
+                
+                return sql.fragment`(select ${spec.sqlAggregateWrap(
+                  sql.fragment`(${sql.identifier(
+                    proc.namespaceName,
+                    proc.name
+                  )}(${tableAlias}))`
+                )} from ${sql.identifier(
+                  table.namespaceName,
+                  table.name
+                )} ${tableAlias} where (${sql.join(conditions, " AND ")}))`;
```

Comment:
See if this much simpler approach works instead:

```suggestion
                return spec.sqlAggregateWrap(
                  sql.fragment`(${sql.identifier(
                    proc.namespaceName,
                    proc.name
                  )}(${tableAlias}))`
                );
```

Reply:
```
  postgraphile:postgres with __local_0__ as (
  select to_json(
    (
      json_build_object(
        'id'::text,
        ((__local_1__."id"))::text
      )
    )
  ) as "@nodes"
  from (
    select __local_1__.*
    from "core_api"."companies" as __local_1__
    where (__local_1__."is_archived" is false)
    and (
      (
        (
          (
            select (
              (
                (
                  (
                    avg(
                      "core_api"."loads_price_per_mile"(__local_2__)
                    ) IS NOT NULL
                  )
                )
              )
            )
            from (
              select *
              from "core_api"."loads" as __local_2__
              where (__local_2__."broker_id" = __local_1__."id")
              and (true)
            ) as __local_2__
          )
        )
      )
    ) and (TRUE) and (TRUE)
    order by avg(
      (
        "core_api"."loads_price_per_mile"(__local_3__)
      )
    ) DESC,
    __local_1__."id" ASC
    limit 5
  ) __local_1__
),
__local_4__ as (
  select json_agg(
    to_json(__local_0__)
  ) as data
  from __local_0__
)
select coalesce(
  (
    select __local_4__.data
    from __local_4__
  ),
  '[]'::json
) as "data" +5ms
  postgraphile:postgres:error ERROR[42703]: column "__local_3__" does not exist +29s
```

Reply:
I somehow missed that we were dealing with a foreign key relation here - sorry about that!

I'm super uncomfortable with adding this because it's going to be incredibly expensive to execute. I suggest that you build your own plugin that adds this if you need it, you should be able to do so without forking this plugin.

Reply:
I somehow missed that we were dealing with a foreign key relation here - sorry about that!

I'm super uncomfortable with adding this because it's going to be incredibly expensive to execute. I suggest that you build your own plugin that adds this if you need it, you should be able to do so without forking this plugin.

Reply:
I somehow missed that we were dealing with a foreign key relation here - sorry about that!

I'm super uncomfortable with adding this because it's going to be incredibly expensive to execute. I suggest that you build your own plugin that adds this if you need it, you should be able to do so without forking this plugin.